geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -1307,7 +1312,11 @@ SelectableChannel getSelectableChannel() {
   }
 
   void registerWithSelector2(Selector s) throws ClosedChannelException {
-    getSelectableChannel().register(s, SelectionKey.OP_READ, this);
+    if ((getIOFilter() != null) && (getIOFilter() instanceof NioSslEngine)) {
```

Comment:
I think the "!= null" is not needed since "null instanceof X" always returns false.